### ISSUE TITLE
Example of slam and imu fusion? 

### ISSUE THREAD

**Issue by antithing** [jpl-x/x#5]
Hi, thanks for making this code available. I am looking to implement inertial fusion into a slam algorithm, is there any examples here that do that?

Thanks!

**Comment by jeff-delaune** [comment]
Hi,

Yes, there's a ROS implementation available here:

https://github.com/jpl-x/x_vio_ros

And a technical report about this implementation here: 

https://arxiv.org/abs/2103.15215

**Comment by antithing** [comment]
Thank you! I need to run without ROS so I will attempt to modify the code. The matchesptr object, is that the 2d features, as pixel location?

**Comment by jeff-delaune** [comment]
Yes, the matches_ptr includes the feature matches from to the current images in pixel coordinates (+ a placeholder for the 3D coordinates for debugging purposes). For more details, the import code is located in line 339 at https://github.com/jpl-x/x/blob/master/src/x/vio/vio.cpp

Please note the match_ptr is only used if you decide to use an external feature tracker (or simulation) that sends data in this format. If you use the image callback instead, xVIO will run its own feature tracker.
